### PR TITLE
12956 ProGuard rules

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://swrvedev.jira.com/browse/SWRVE-12956
Original PR here: https://github.com/Swrve/swrve-android-sdk/pull/199 (thanks Silin!)
(Confirmed that the pro guard file ends up in the .aar package)

Used recommendations from https://github.com/google/gson/blob/master/examples/android-proguard-example/proguard.cfg

### CODE REVIEW

**Comment by Sergio-Mira** [comment]
@dominicmarmionswrve Could you have a quick look?

**Comment by dominicmarmionswrve** [comment]
🚀

File: SwrveConversationSDK/proguard-rules.pro [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# Do not obfuscate fields of model classes to be parsed by GSON
--keepclassmembernames public class com.swrve.sdk.conversations.engine.model.** {
-   <fields>;
-}
+# GSON proguard config
```
Comment (dominicmarmionswrve) [inline SwrveConversationSDK/proguard-rules.pro:1]: 👍

File: SwrveSDK/proguard-rules.pro [hunk 2]
```diff
@@ -1,4 +1,6 @@
-# Do not obfuscate fields of model classes to be parsed by GSON
--keepclassmembernames public class com.swrve.sdk.messaging.model.** {
-   <fields>;
-}
+# GSON proguard config
```
Comment (dominicmarmionswrve) [inline SwrveSDK/proguard-rules.pro:1]: 👍

**Pull request merged**